examples/large_models/inferentia2/llama2/inf2_handler.py

```diff
@@ -0,0 +1,100 @@
+
+
+class LLMHandler(BaseHandler, ABC):
+    """
+    Transformers handler class for text completion streaming on Inferentia2
+    """
+
+    def __init__(self):
+        super(LLMHandler, self).__init__()
+        self.initialized = False
+
+    def initialize(self, ctx):
+        self.manifest = ctx.manifest
+        properties = ctx.system_properties
+        model_dir = properties.get("model_dir")
+
+        # settings for model compiliation and loading
+        model_name = ctx.model_yaml_config["handler"]["model_name"]
+        tp_degree = ctx.model_yaml_config["handler"]["tp_degree"]
+        self.max_length = ctx.model_yaml_config["handler"]["max_length"]
+
+        # allocate "tp_degree" number of neuron cores to the worker process
+        os.environ["NEURON_RT_NUM_CORES"] = str(tp_degree)
+        try:
+            num_neuron_cores_available = (
+                torch_neuronx.xla_impl.data_parallel.device_count()
+            )
+            assert num_neuron_cores_available >= int(tp_degree)
+        except (RuntimeError, AssertionError) as error:
+            raise RuntimeError(
+                "Required number of neuron cores for tp_degree "
+                + str(tp_degree)
+                + " are not available: "
+                + str(error)
+            )
+
+        os.environ["NEURON_CC_FLAGS"] = "--model-type=transformer-inference"
+
+        self.tokenizer = LlamaTokenizer.from_pretrained(model_name)
+        self.model = LlamaForSampling.from_pretrained(
+            model_dir, batch_size=1, tp_degree=tp_degree
+        )
+        logger.info("Starting to compile the model")
+        self.model.to_neuron()
+        logger.info("Model has been successfully compiled")
+        model_config = AutoConfig.from_pretrained(model_dir)
+        self.model = HuggingFaceGenerationModelAdapter(model_config, self.model)
+        self.output_streamer = TextIteratorStreamer(self.tokenizer)
+
+        self.initialized = True
+
+    def preprocess(self, requests):
+        input_texts = []
+        for req in requests:
+            data = req.get("data") or req.get("body")
+            if isinstance(data, (bytes, bytearray)):
+                data = data.decode("utf-8")
+            input_texts.append(data)
+
+        return self.tokenizer(input_texts, return_tensors="pt")
+
+    def inference(self, tokenized_input):
+        generation_kwargs = dict(
+            tokenized_input,
+            streamer=self.output_streamer,
+            max_new_tokens=self.max_length,
+        )
+        self.model.reset_generation()
+        thread = Thread(target=self.model.generate, kwargs=generation_kwargs)
+        thread.start()
+
+        for new_text in self.output_streamer:
+            send_intermediate_predict_response(
+                [new_text],
+                self.context.request_ids,
+                "Intermediate Prediction success",
+                200,
+                self.context,
+            )
+
+        thread.join()
```

Comment:
Tested with batch size 2 and got the following:
```
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/torch/autograd/grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "/opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/transformers/generation/utils.py", line 1363, in generate
    streamer.put(input_ids.cpu())
  File "/opt/aws_neuron_venv_pytorch/lib/python3.8/site-packages/transformers/generation/streamers.py", line 87, in put
    raise ValueError("TextStreamer only supports batch size 1")
ValueError: TextStreamer only supports batch size 1
```

Since `TextStreamer` only supports batch size 1, I'll update the `preprocess` functions to handle only batch size 1 and raise exception if higher batch size input is received.